examples/step-74/step-laplace-FEIV.cc

```diff
@@ -0,0 +1,1245 @@
+                                const unsigned int &face_no, 
+                                ScratchData1<dim> &scratch_data, 
+                                CopyData &copy_data)
+    {
+      scratch_data.fe_interface_values.reinit(cell, face_no);
+      const FEFaceValuesBase<dim> &fe_fv =
+      scratch_data.fe_interface_values.get_fe_face_values(0);
+
+      const auto &q_points = fe_fv.get_quadrature_points();
+      const unsigned n_q_points = q_points.size();
+      
+      const std::vector<double> &JxW = fe_fv.get_JxW_values ();
+      
+      std::vector<double> nu (n_q_points);
+      viscosity_function.value_list (q_points, nu);
+
+      std::vector<double> g(n_q_points);
+      boundary_function.value_list (q_points, g);
+
+      std::vector<double> sol_u(n_q_points);
+      fe_fv.get_function_values(solution, sol_u);
+
+      const double degree = std::max(1.0, static_cast<double> (fe_fv.get_fe().degree));
+      const double extent1 = cell->extent_in_direction(GeometryInfo<dim>::unit_normal_direction[face_no]);
+      const double penalty = 4.0 * degree * (degree+1.0) / (extent1);
+     
+      double difference_norm_square = 0;
+      for (unsigned int point=0; point<q_points.size(); ++point)
+        {
+          const double diff= (g[point]-sol_u[point]);
+          difference_norm_square += diff*diff*JxW[point];
+        }
+        copy_data.value +=std::sqrt(penalty*difference_norm_square);
+    };
+
+    auto face_worker = [&] (const Iterator &cell, 
+                            const unsigned int &f, 
+                            const unsigned int &sf,
+                            const Iterator &ncell, 
+                            const unsigned int &nf,
+                            const unsigned int &nsf,
+                            ScratchData1<dim> &scratch_data,
+                            CopyData &copy_data)
+    {
+      FEInterfaceValues<dim> &fe_iv = scratch_data.fe_interface_values;
+      fe_iv.reinit(cell, f, sf, ncell, nf, nsf);  
+      
+      copy_data.face_data.emplace_back();
+      CopyDataFace &copy_data_face = copy_data.face_data.back();
+      
+      copy_data_face.cell_indices[0] = cell->active_cell_index();
+      copy_data_face.cell_indices[1] = ncell->active_cell_index();
+
+      const std::vector<double> &JxW = fe_iv.get_JxW_values ();
+      const std::vector<Tensor<1,dim> > &normals = fe_iv.get_normal_vectors ();
+      
+      const auto &q_points = fe_iv.get_quadrature_points();
+      const unsigned int n_q_points = q_points.size();
+
+      std::vector<double> nu (n_q_points);
+      viscosity_function.value_list (q_points, nu);
+      
+      std::vector<Tensor<1, dim>> grad_u[2];
+      std::vector<double> sol_u[2];
+
+      const double h = cell->face(f)->measure();
+      for(unsigned int i = 0; i < 2; ++i)
+      {
+        grad_u[i].resize(n_q_points);
+        sol_u[i].resize(n_q_points);
+        fe_iv.get_fe_face_values(i).get_function_values(
```

Comment:
okay, that makes sense, but I think this needs to go into FEInterfaceValues.

Reply:
fe_iv.get_function_jump(solution, std::vector<double> values)

or

fe_iv.get_function_values(solution, std::vector<double>[2] values)